marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/dhs/updateResourcePermissions.sjs

```diff
@@ -0,0 +1,39 @@
+/*
+  Copyright 2012-2019 MarkLogic Corporation
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+'use strict';
+
+var extensionType;
+var fileName;
+
+
+xdmp.eval('let partUri = "/marklogic.rest.'+ extensionType + '/' + fileName + '/assets/"; '+
```

Comment:
This code can all be reused in the Java command - though I don't think you need an xdmp.eval here. Just do what CopyQueryOptionsCommand is doing - call hubConfig.newModulesDbClient().